models/recurly__churn_analysis.sql

```diff
@@ -1,18 +1,18 @@
 with subscription_churn_reason as 
 (
     select *,
-         case when account_state not like 'active' then 'Account Closed'
-            when expiration_reason like 'canceled' then 'Canceled'
-            when expiration_reason like 'nonpayment_trial' then 'Trial Ended'
-            when expiration_reason like 'nonpayment_gift' then 'Gift Ended'
-            when expiration_reason like 'nonpayment' then 'Non-payment'
-            when expiration_reason like '%tax%' then 'Tax Location Invalid'
-            when has_auto_renew is false then 'Non-renewing'
+         case when account_state not like 'active' then 'Account Closed' 
+            when expiration_reason like 'canceled' then 'Canceled' 
+            when expiration_reason like 'nonpayment_trial' then 'Trial Ended' 
+            when expiration_reason like 'nonpayment_gift' then 'Gift Ended' 
+            when expiration_reason like 'nonpayment' then 'Non-payment' 
```

Comment:
Yep, all good feedback and changing up! 

I think the big reason I used like commands is that I'm still not exactly certain what the format looks like for some of these expiration reasons, so leaned more toward string commands. Although I didn't even add %% to most of them, so removing for the time being and will revise once we have a larger data set to draw from.

Reply:
@fivetran-joemarkiewicz Sounds like a good idea! Do you have a good example of a for loop/dictionary variable we've created in the past? Not super familiar with implementing either.

Reply:
We actually do not use the dictionary/for loop combination in any of our existing packages.... However, it actually looks like I documented how this can be done when I was experimenting with it in our QuickBooks package (but never actually implemented it). 

Past Joe to the rescue haha you can use [this Slab post](https://fivetran.slab.com/posts/miscellaneous-solutions-team-relevant-notes-r83ebv8j#hf73b-unpacking-a-dictionary-with-a-for-loop-using-dbt) as a reference for how to add this type of functionality to the model.